flight/Modules/Battery/battery.c

```diff
@@ -128,17 +128,32 @@ static void batteryTask(void * parameters)
 				currentADCPin = -1;
 		}
 
+		bool adc_pin_invalid = false;
+		bool adc_offset_invalid = false;
+
 		// handle voltage
 		if (voltageADCPin >= 0) {
-			flightBatteryData.Voltage = (float)PIOS_ADC_GetChannelVolt(voltageADCPin);
-
-			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns -1 on error)
-			if(flightBatteryData.Voltage < 0.0f)
-				flightBatteryData.Voltage = 0.0f;
-
-			// scale to actual voltage
-			flightBatteryData.Voltage *= 1000.0f / batterySettings.SensorCalibrationFactor[FLIGHTBATTERYSETTINGS_SENSORCALIBRATIONFACTOR_VOLTAGE];
-			flightBatteryData.Voltage += batterySettings.SensorCalibrationOffset[FLIGHTBATTERYSETTINGS_SENSORCALIBRATIONOFFSET_VOLTAGE]; //in Volts
+			float adc_voltage = (float)PIOS_ADC_GetChannelVolt(voltageADCPin);
+			float scaled_voltage = 0.0f;
+
+			// A negative result indicates an error (PIOS_ADC_GetChannelVolt returns negative on error)
+			if(adc_voltage < 0.0f)
+				adc_pin_invalid = true;
+			else {
+				// scale to actual voltage
+				scaled_voltage = adc_voltage * 1000.0f
```

Comment:
What would you say about putting a parentheses around this, so it's 
```
... = (adc_voltage * 1000.0f 
   / batterySettings.SensorCalibrationFactor[FLIGHTBATTERYSETTINGS_SENSORCALIBRATIONFACTOR_VOLTAGE])
   +...
```

That might make the code less brittle in case someone moves things around and accidentally messes up the order of operations.